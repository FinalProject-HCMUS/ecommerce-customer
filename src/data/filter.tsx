export const categories = [
  { id: 'tshirts', label: 'T-shirts' },
  { id: 'shorts', label: 'Shorts' },
  { id: 'shirts', label: 'Shirts' },
  { id: 'hoodies', label: 'Hoodies' },
  { id: 'jeans', label: 'Jeans' },
];

export const colors = [
  { id: 'green', color: '#4CAF50' },
  { id: 'red', color: '#F44336' },
  { id: 'yellow', color: '#FFEB3B' },
  { id: 'orange', color: '#FF9800' },
  { id: 'blue', color: '#2196F3' },
  { id: 'purple', color: '#9C27B0' },
  { id: 'pink', color: '#E91E63' },
  { id: 'white', color: '#FFFFFF', border: true },
  { id: 'black', color: '#000000' },
];

export const sizes = [
  { id: 'xxs', label: 'XX-Small' },
  { id: 'xs', label: 'X-Small' },
  { id: 'small', label: 'Small' },
  { id: 'medium', label: 'Medium' },
  { id: 'large', label: 'Large' },
  { id: 'xl', label: 'X-Large' },
  { id: 'xxl', label: 'XX-Large' },
  { id: '3xl', label: '3X-Large' },
  { id: '4xl', label: '4X-Large' },
];
