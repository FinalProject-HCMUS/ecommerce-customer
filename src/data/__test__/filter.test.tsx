import { categories, colors, sizes } from '../filter';

describe('filter data', () => {
  test('categories should be an array with correct items', () => {
    expect(Array.isArray(categories)).toBe(true);
    expect(categories.length).toBeGreaterThan(0);
    categories.forEach(cat => {
      expect(cat).toHaveProperty('id');
      expect(cat).toHaveProperty('label');
      expect(typeof cat.id).toBe('string');
      expect(typeof cat.label).toBe('string');
    });
    expect(categories.map(c => c.id)).toEqual(
      expect.arrayContaining(['tshirts', 'shorts', 'shirts', 'hoodies', 'jeans'])
    );
  });

  test('colors should be an array with correct items', () => {
    expect(Array.isArray(colors)).toBe(true);
    expect(colors.length).toBeGreaterThan(0);
    colors.forEach(color => {
      expect(color).toHaveProperty('id');
      expect(color).toHaveProperty('color');
      expect(typeof color.id).toBe('string');
      expect(typeof color.color).toBe('string');
    });
    // Check for a color with border property
    expect(colors.find(c => c.id === 'white')).toHaveProperty('border', true);
  });

  test('sizes should be an array with correct items', () => {
    expect(Array.isArray(sizes)).toBe(true);
    expect(sizes.length).toBeGreaterThan(0);
    sizes.forEach(size => {
      expect(size).toHaveProperty('id');
      expect(size).toHaveProperty('label');
      expect(typeof size.id).toBe('string');
      expect(typeof size.label).toBe('string');
    });
    expect(sizes.map(s => s.id)).toEqual(
      expect.arrayContaining(['xxs', 'xs', 'small', 'medium', 'large', 'xl', 'xxl', '3xl', '4xl'])
    );
  });

  test('should match snapshot', () => {
    expect({ categories, colors, sizes }).toMatchSnapshot();
  });
});