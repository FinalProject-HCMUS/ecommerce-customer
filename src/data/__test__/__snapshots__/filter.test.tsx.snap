// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter data should match snapshot 1`] = `
{
  "categories": [
    {
      "id": "tshirts",
      "label": "T-shirts",
    },
    {
      "id": "shorts",
      "label": "Shorts",
    },
    {
      "id": "shirts",
      "label": "Shirts",
    },
    {
      "id": "hoodies",
      "label": "Hoodies",
    },
    {
      "id": "jeans",
      "label": "Jeans",
    },
  ],
  "colors": [
    {
      "color": "#4CAF50",
      "id": "green",
    },
    {
      "color": "#F44336",
      "id": "red",
    },
    {
      "color": "#FFEB3B",
      "id": "yellow",
    },
    {
      "color": "#FF9800",
      "id": "orange",
    },
    {
      "color": "#2196F3",
      "id": "blue",
    },
    {
      "color": "#9C27B0",
      "id": "purple",
    },
    {
      "color": "#E91E63",
      "id": "pink",
    },
    {
      "border": true,
      "color": "#FFFFFF",
      "id": "white",
    },
    {
      "color": "#000000",
      "id": "black",
    },
  ],
  "sizes": [
    {
      "id": "xxs",
      "label": "XX-Small",
    },
    {
      "id": "xs",
      "label": "X-Small",
    },
    {
      "id": "small",
      "label": "Small",
    },
    {
      "id": "medium",
      "label": "Medium",
    },
    {
      "id": "large",
      "label": "Large",
    },
    {
      "id": "xl",
      "label": "X-Large",
    },
    {
      "id": "xxl",
      "label": "XX-Large",
    },
    {
      "id": "3xl",
      "label": "3X-Large",
    },
    {
      "id": "4xl",
      "label": "4X-Large",
    },
  ],
}
`;
